### PR TITLE
Add network policy enforcement latency measurement to load test

### PR DESCRIPTION
**Pull request by dlapcevic**
Integrate network policy enforcement latency measurement (https://github.com/kubernetes/perf-tests/pull/2234) with the CL2 load test.

/kind feature

### CODE REVIEW

File: clusterloader2/testing/load/deployment.yaml [hunk 1]
```diff
@@ -28,18 +36,22 @@ spec:
...
+  {{if $USE_ADVANCED_DNSTEST}}
       - image: gcr.io/k8s-staging-perf-tests/dnsperfgo:v1.3.0
         ports:
         - containerPort: 9153
           name: dnsperfmetrics
           protocol: TCP
-{{else}}
+  {{else}}
       - image: gcr.io/k8s-staging-perf-tests/dnsperfgo:v1.2.0
 {{end}}
```
Comment (tosi3k) [inline clusterloader2/testing/load/deployment.yaml:56]: If we change the indentation, please indent this line as well for consistency's sake.

**Review by tosi3k — CHANGES_REQUESTED** [review]

File: clusterloader2/testing/load/modules/network-policy/net-policy-enforcement-latency.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+{{$setup := DefaultParam .setup false}}
+{{$run := DefaultParam .run false}}
+{{$complete := DefaultParam .complete false}}
+{{$testType := DefaultParam .testType "policy-creation"}}
+# Target port needs to match the server container port of target pods that have
+# "targetLabelKey: targetLabelValue" label selector.
+{{$targetPort := 80}}
+
+{{if $ENABLE_NETWORK_POLICY_ENFORCEMENT_LATENCY_TEST}}
```
[the remark this answers is not in the document]
Comment (dlapcevic) [reply clusterloader2/testing/load/modules/network-policy/net-policy-enforcement-latency.yaml]: Done

File: clusterloader2/testing/load/modules/network-policy/net-policy-metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
+{{$ENABLE_NETWORK_POLICY_ENFORCEMENT_LATENCY_TEST := DefaultParam .CL2_ENABLE_NETWORK_POLICY_ENFORCEMENT_LATENCY_TEST false}}
+
+# Valid actions: "start", "gather"
+{{$action := .action}}
+{{$usePolicyCreationMetrics := DefaultParam .usePolicyCreationMetrics true}}
+{{$usePodCreationMetrics := DefaultParam .usePodCreationMetrics true}}
+{{$useCiliumMetrics := DefaultParam .useCiliumMetrics true}}
+
+{{if $ENABLE_NETWORK_POLICY_ENFORCEMENT_LATENCY_TEST}}
```
[the remark this answers is not in the document]
Comment (dlapcevic) [reply clusterloader2/testing/load/modules/network-policy/net-policy-metrics.yaml]: Done

File: clusterloader2/testing/load/modules/network-policy/net-policy-metrics.yaml [hunk 4]
```diff
@@ -0,0 +1,98 @@
+{{$ENABLE_NETWORK_POLICY_ENFORCEMENT_LATENCY_TEST := DefaultParam .CL2_ENABLE_NETWORK_POLICY_ENFORCEMENT_LATENCY_TEST false}}
```
[the remark this answers is not in the document]
Comment (dlapcevic) [reply clusterloader2/testing/load/modules/network-policy/net-policy-metrics.yaml]: Done.

**Comment by dlapcevic** [comment]
I resolved the comments and added thresholds to the `GenericPrometheusQuery` in `net-policy-metrics.yaml`.

**Comment by tosi3k** [comment]
/lgtm
/hold

Feel free to unhold after you test the latest version of this PR in a manual run.

**Comment by dlapcevic** [comment]
Latest run passed as well.

/hold cancel

**Pull request merged**